src/book/02-system/01-nubots/03-nuclear.mdx

```diff
@@ -5,3 +5,144 @@ title: NUClear
+| ------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| UDP     | Emits data as a UDP packet over the network. Is of the form `emit<Scope::UDP>(data, to_addr, to_port);`.                             |
+| NETWORK | Emits data over the network to other NUClear environments. Is of the form `emit<Scope::NETWORK>(data, target, reliable, dataType);`. |
+
+# NUClear Roles
+
+The NUClear roles system is a build and messaging system for the NUClear framework. It uses [CMake](https://cmake.org/) and [Python](https://www.python.org/) to manage the construction of various executables made up of a selection of modules. These executables are called roles.
+
+CMake is used as the main system for generating the libraries and executables that are used for the final binary. Note that it utilises [globbing](<https://en.wikipedia.org/wiki/Glob_(programming)>) to find the sources that are used for modules. If you add or remove a file, you must rerun CMake to locate the new files.
+
+# Managing Reactions
+
+During system runtime, executing reactions can be managed via their associated reaction handles. A reaction handle is provided for binding on statements. Once an on statement has been bound, the reaction handle will be enabled. If necessary, reactions can toggle between enabled and disabled during runtime.
+
+| Handle           | Description                                                                                                                                                                                         |
+| ---------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| disable()        | Disables the reaction. Tasks will not be created if triggered.                                                                                                                                      |
+| enable()         | Enables the reaction so that associated tasks will be scheduled and queued when the reaction is triggered.                                                                                          |
+| enable(bool set) | Sets the run status of the reaction handle. `true` will enable the reaction and `false` will disable it.                                                                                                   |
+| enabled()        | Determines the current status of the reaction. Returns `true` if the reaction is enabled, and `false` otherwise.                                                                                                           |
+| unbind()         | Removes a reaction request from the runtime environment. This action is not reversible, once a reaction has been unbound, it is no longer available for further use during that runtime. |
+
+# Examples
```

Comment:
We should probably have a section on Neutrons as well and how they work with messaging

Reply:
What are Neutrons?